frame/proxy/src/lib.rs

```diff
@@ -265,10 +287,14 @@ pub mod pallet {
 		#[pallet::call_index(3)]
 		#[pallet::weight(T::WeightInfo::remove_proxies(T::MaxProxies::get()))]
 		pub fn remove_proxies(origin: OriginFor<T>) -> DispatchResult {
+			// Ensure that the origin is a signed account.
 			let who = ensure_signed(origin)?;
-			let (_, old_deposit) = Proxies::<T>::take(&who);
-			T::Currency::unreserve(&who, old_deposit);
 
+			// Call the `remove_all_proxy_delegates` method to remove all proxies associated with
+			// `who`.
+			<Pallet<T>>::remove_all_proxy_delegates(who)?;
+
+			// Return success.
```

Comment:
The commenting here is a bit excessive -- same with the function body of `remove_all_proxy_delegates`.

Reply:
removed